x/perp/v2/keeper/clearing_house.go

```diff
@@ -555,6 +555,11 @@ func (k Keeper) afterPositionUpdate(
 
 	if !positionResp.Position.Size_.IsZero() {
 		k.Positions.Insert(ctx, collections.Join(market.Pair, traderAddr), positionResp.Position)
+	} else {
+		if !positionResp.Position.Margin.IsZero() {
```

Comment:
Do we ever zero out the position's `margin`? I think we keep it there when we reverse a position.